From c42734bcb8564ed72ccf1d5a61605e8db4cc676a Mon Sep 17 00:00:00 2001
From: "M, Kumar K" <kumar.k.m@intel.com>
Date: Thu, 27 Dec 2018 12:06:49 +0530
Subject: [PATCH] Allow virtual driver to be used as primary

Revert "Set available device as prmary hardware device"

This reverts commit a1506d062fa75e7cdee46d908d3cc103e5727716.

Change-Id: I662b390ef224b0fcb4542c509f105bd528cd53d6
Tracked-On: OAM-OAM-73881
Signed-off-by: M, Kumar K <kumar.k.m@intel.com>
---
 services/audioflinger/AudioFlinger.cpp | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/services/audioflinger/AudioFlinger.cpp b/services/audioflinger/AudioFlinger.cpp
index 2eef169..bdd39c6 100644
--- a/services/audioflinger/AudioFlinger.cpp
+++ b/services/audioflinger/AudioFlinger.cpp
@@ -2159,22 +2159,14 @@ status_t AudioFlinger::openOutput(audio_module_handle_t module,
                 mHardwareStatus = AUDIO_HW_SET_MODE;
                 mPrimaryHardwareDev->hwDevice()->setMode(mMode);
                 mHardwareStatus = AUDIO_HW_IDLE;
-            } else if (mPrimaryHardwareDev == NULL)  {
-                ALOGV("Using available module %d as the primary audio interface", module);
-                mPrimaryHardwareDev = playbackThread->getOutput()->audioHwDev;
-
-                AutoMutex lock(mHardwareLock);
-                mHardwareStatus = AUDIO_HW_SET_MODE;
-                mPrimaryHardwareDev->hwDevice()->setMode(mMode);
-                mHardwareStatus = AUDIO_HW_IDLE;
             }
-
         } else {
             MmapThread *mmapThread = (MmapThread *)thread.get();
             mmapThread->ioConfigChanged(AUDIO_OUTPUT_OPENED);
         }
         return NO_ERROR;
     }
+
     return NO_INIT;
 }
 
-- 
1.9.1

